File: small_problems/easy2/sum_or_product_of_consecutive_integers.rb
def prompt(message)
  puts ">> #{message}"
end

prompt("Please enter an integer greater than 0:")
operand = gets.chomp.to_i

prompt("Enter 's' to compute the sum, 'p' to compute the product.")
operation = gets.chomp

result = case operation
        when 's'
          (1..operand).inject(:+) # can also use (1..n).each { |i| total += i }, need to initiate total in this case
        else
          (1..operand).inject(:*)
        end

operator = operation == 's' ? 'sum' : 'product'

puts "The #{operator} of the integers between 1 and #{operand} is #{result}."

File: javascript_small_problems/easy1/sum_or_product_of_consecutive_integers.js
/*
Please enter an integer greater than 0: 5
Enter "s" to compute the sum, or "p" to compute the product. s

The sum of the integers between 1 and 5 is 15.

Please enter an integer greater than 0: 6
Enter "s" to compute the sum, or "p" to compute the product. p

The product of the integers between 1 and 6 is 720.
*/

let rlSync = require('readline-sync');
let integer = parseInt(rlSync.question('Please enter an integer greater than 0: '));
let type = rlSync.question('Enter "s" to compute the sum, or "p" to compute the product: ');

let compute = 1;
switch (type) {
  case 's':
    for (let current = 2; current <= integer; current += 1) {
      compute += current;
    }
    break;
  case 'p':
    for (let current = 2; current <= integer; current += 1) {
      compute *= current;
    }
    break;
}

console.log(`The ${type === 's' ? 'sum' : 'product'} of the integers between 1 and ${integer} is ${compute}.`);
